src/pages/preferences/preferencesTabs/publicKeysTab/PublicKeysTable.tsx

```diff
@@ -0,0 +1,133 @@
+import { useRef } from "react";
+import { useQuery, useMutation } from "@apollo/client";
+import styled from "@emotion/styled";
+import Button from "@leafygreen-ui/button";
+import Icon from "@leafygreen-ui/icon";
+import { useLeafyGreenTable } from "@leafygreen-ui/table";
+import { usePreferencesAnalytics } from "analytics";
+import Popconfirm from "components/Popconfirm";
+import { WordBreak } from "components/styles";
+import { BaseTable } from "components/Table/BaseTable";
+import { TablePlaceholder } from "components/Table/TablePlaceholder";
+import { size } from "constants/tokens";
+import { useToastContext } from "context/toast";
+import {
+  MyPublicKeysQuery,
+  MyPublicKeysQueryVariables,
+  PublicKey,
+  RemovePublicKeyMutation,
+  RemovePublicKeyMutationVariables,
+} from "gql/generated/types";
+import { REMOVE_PUBLIC_KEY } from "gql/mutations";
+import { MY_PUBLIC_KEYS } from "gql/queries";
+import { EditModalPropsState } from "./EditModal";
+
+type PublicKeysTableProps = {
+  setEditModalProps: React.Dispatch<React.SetStateAction<EditModalPropsState>>;
+};
+
+export const PublicKeysTable: React.FC<PublicKeysTableProps> = ({
+  setEditModalProps,
+}) => {
+  const dispatchToast = useToastContext();
+  const { sendEvent } = usePreferencesAnalytics();
+
+  const { data: myKeysData, loading: loadingMyPublicKeys } = useQuery<
+    MyPublicKeysQuery,
+    MyPublicKeysQueryVariables
+  >(MY_PUBLIC_KEYS, {
+    onError(error) {
+      dispatchToast.error(
+        `There was an error fetching your public keys: ${error.message}`
+      );
+    },
+  });
+
+  const [removePublicKey, { loading: loadingRemovePublicKey }] = useMutation<
+    RemovePublicKeyMutation,
+    RemovePublicKeyMutationVariables
+  >(REMOVE_PUBLIC_KEY, {
+    onError(error) {
+      dispatchToast.error(
+        `There was an error removing the public key: ${error.message}`
+      );
+    },
+    refetchQueries: ["MyPublicKeys"],
+  });
+
+  const columns = [
+    {
+      header: "Name",
+      accessorKey: "name",
+      cell: ({ getValue }) => (
+        <WordBreak data-cy="table-key-name">{getValue()}</WordBreak>
+      ),
+    },
+    {
+      header: "Actions",
+      cell: ({ row }) => {
+        const { key, name } = row.original;
+        return (
+          <ButtonContainer className="w-[100px]">
+            <Button
+              size="small"
+              data-cy="edit-btn"
+              leftGlyph={<Icon glyph="Edit" />}
+              onClick={() => {
+                setEditModalProps({
+                  initialPublicKey: { key, name },
+                  visible: true,
+                });
+              }}
+            />
```

Comment:
Possibly could look nice if we turned these into `IconButton`s? its just a suggestion :)

Reply:
Alas it looked a bit weird :( we use this action button pattern elsewhere too so I think the consistency is good!